Lib/test/test_concurrent_futures.py

```diff
@@ -728,24 +727,29 @@ def test_no_timeout(self):
                  future1, future2]),
                 completed)
 
-    def test_zero_timeout(self):
-        future1 = self.executor.submit(time.sleep, 2)
-        completed_futures = set()
-        try:
-            for future in futures.as_completed(
-                    [CANCELLED_AND_NOTIFIED_FUTURE,
-                     EXCEPTION_FUTURE,
-                     SUCCESSFUL_FUTURE,
-                     future1],
-                    timeout=0):
-                completed_futures.add(future)
-        except futures.TimeoutError:
-            pass
+    def test_future_times_out(self):
+        """Test ``futures.as_completed`` timing out before
+        completing it's final future."""
+        already_completed = {CANCELLED_AND_NOTIFIED_FUTURE,
+                             EXCEPTION_FUTURE,
+                             SUCCESSFUL_FUTURE}
 
-        self.assertEqual(set([CANCELLED_AND_NOTIFIED_FUTURE,
-                              EXCEPTION_FUTURE,
-                              SUCCESSFUL_FUTURE]),
-                         completed_futures)
+        for timeout in (0, 0.01):
+            with self.subTest(timeout):
+
+                future = self.executor.submit(time.sleep, 0.1)
```

Comment:
The test *actually* sleeps this long (possibly when the thread is torn down?) so reducing the time from 2s is quite beneficial for test runtime. Particularly so as this test is expanded to run multiple times:

2s sleep:

![image](https://user-images.githubusercontent.com/48555120/209476520-d3cb183c-b788-4952-b9ff-abad44f1c774.png)

0.1s sleep:
![image](https://user-images.githubusercontent.com/48555120/209476510-3c1d0c52-1dda-4639-a551-d6ff27fcd2a3.png)

----------------------------------------------------
Python + pytest on 3.12 with the command:
`./python -m pytest Lib/test/test_concurrent_futures.py -k test_future_times_out -vvv`

Reply:
If this winds up flaky due to timing issues on CI systems or buildbots we can increase it. They're often under heavy load, so concurrency tests depending on one thread executing more than another can be flaky.